### ISSUE TITLE
Uncaught TypeError: Cannot read property 'split' of undefined

### ISSUE THREAD

**Issue by edrubin** [mark-hahn/markdown-scroll-sync#45]
[Enter steps to reproduce below:]

1. Open Atom
2. Create new file

**Atom Version**: 1.4.3
**System**: Mac OS X 10.11.3
**Thrown From**: [markdown-scroll-sync](https://github.com/mark-hahn/markdown-scroll-sync) package, v2.1.1

### Stack Trace

Uncaught TypeError: Cannot read property 'split' of undefined

```
At /Users/edwardarubin/.atom/packages/markdown-scroll-sync/lib/main.coffee:29

TypeError: Cannot read property 'split' of undefined
    at isMarkdown (/Users/edwardarubin/.atom/packages/markdown-scroll-sync/lib/main.coffee:29:46)
    at /Users/edwardarubin/.atom/packages/markdown-scroll-sync/lib/main.coffee:33:10
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/pane-container.js:333:27
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at Pane.module.exports.Pane.setActiveItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:248:22)
    at Pane.module.exports.Pane.activateItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:316:21)
    at Pane.module.exports.Pane.activateItemAtIndex (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:310:19)
    at Pane.module.exports.Pane.activateNextItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:267:21)
    at Pane.module.exports.Pane.removeItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:389:16)
    at Pane.module.exports.Pane.destroyItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:453:16)
    at Pane.module.exports.Pane.destroyActiveItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:433:12)
    at Workspace.module.exports.Workspace.destroyActivePaneItem (/Applications/Atom.app/Contents/Resources/app.asar/src/workspace.js:656:35)
    at Workspace.module.exports.Workspace.closeActivePaneItemOrEmptyPaneOrWindow (/Applications/Atom.app/Contents/Resources/app.asar/src/workspace.js:695:21)
    at atom-workspace.commandRegistry.add.core:close (/Applications/Atom.app/Contents/Resources/app.asar/src/register-default-commands.js:170:32)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:536:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:359:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeydown (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:97:36)
    at HTMLDocument.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
  2x -0:07.7.0 application:new-file (atom-pane.pane.active)
  3x -0:02.6.0 core:close (atom-pane.pane.active)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "language-markdown",
      "markdown-preview",
      "block-cursor",
      "terminal-plus",
      "antialiased",
      "omni-ruler"
    ],
    "ignoredNames": [
      ".git",
      ".svn",
      ".DS_Store"
    ],
    "openEmptyEditorOnStart": false,
    "packagesWithKeymapsDisabled": [
      "mathjax-wrapper"
    ],
    "themes": [
      "native-ui",
      "proton-kai"
    ]
  }
}
```

### Installed Packages

```coffee
# User
Sublime-Style-Column-Selection, v1.5.1
atom-clock, v0.0.6
autocomplete-python, v1.6.1
color-picker, v2.1.1
cursor-blink-interval, v0.3.1
cursor-indicator, v1.2.7
file-icons, v1.6.14
highlight-selected, v0.11.2
ink, v0.3.2
language-julia, v0.5.0
language-latex, v0.6.1
language-r, v0.4.0
language-stata, v0.1.0
latex, v0.29.0
latextools, v0.7.6
markdown-preview-plus, v2.2.2
markdown-scroll-sync, v2.1.1
minimap, v4.19.0
minimap-cursorline, v0.1.0
native-ui, v0.17.0
pdf-view, v0.39.0
pigments, v0.23.1
proton-kai, v0.1.0
wordcount, v2.6.0

# Dev
No dev packages
```

**Comment by silverskyvicto** [comment]
[Enter steps to reproduce below:]

1. Open Atom
2. Create New File

**Atom Version**: 1.5.0
**System**: Mac OS X 10.11.3
**Thrown From**: [markdown-scroll-sync](https://github.com/mark-hahn/markdown-scroll-sync) package, v2.1.1


### Stack Trace

Uncaught TypeError: Cannot read property 'split' of undefined

```
At /Users/ey/.atom/packages/markdown-scroll-sync/lib/main.coffee:29

TypeError: Cannot read property 'split' of undefined
    at isMarkdown (/Users/ey/.atom/packages/markdown-scroll-sync/lib/main.coffee:29:46)
    at /Users/ey/.atom/packages/markdown-scroll-sync/lib/main.coffee:33:10
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/pane-container.js:333:27
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at Pane.module.exports.Pane.setActiveItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:248:22)
    at Pane.module.exports.Pane.activateItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:327:21)
    at atom-tabs.TabBarView.onMouseDown (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:670:19)
```

### Commands

```
     -1:32.2.0 application:new-file (atom-text-editor.editor.is-focused)
     -1:23.4.0 tree-view:reveal-active-file (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-one-dark-syntax.theme-one-dark-ui)
     -1:18.3.0 application:new-file (atom-text-editor.editor.is-focused)
  8x -1:13.1.0 core:backspace (atom-text-editor.editor.is-focused)
     -1:09.4.0 tree-view:toggle (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "language-gfm",
      "linter-markdown",
      "whitespace",
      "node-resolver",
      "autohide-tree-view",
      "java-generator",
      "autocomplete-html-entities"
    ]
  }
}
```

### Installed Packages

```coffee
# User
MavensMate-Atom, v1.0.7
Stylus, v2.1.2
Sublime-Style-Column-Selection, v1.5.1
angularjs, v0.3.4
angularjs-snippets, v0.10.18
api-workbench, v0.8.17
atom-alignment, v0.12.1
atom-django, v0.3.2
atom-html-preview, v0.1.13
atom-jshint, v2.0.0
atom-json-editor, v0.4.1
atom-python-debugger, v0.3.4
atom-ternjs, v0.13.2
autoclose-html, v0.20.0
autocomplete-angular-material, v0.4.0
autocomplete-modules, v1.3.1
autocomplete-paths, v1.0.2
autocomplete-php, v0.3.7
autocomplete-python, v1.6.1
autocomplete-ruby, v0.1.0
browser-plus, v0.0.54
build-cargo, v0.12.0
build-gradle, v0.6.0
color-picker, v2.1.1
css-snippets, v0.9.0
data-atom, v0.18.1
docblockr, v0.7.3
file-icons, v1.6.14
file-type-icons, v0.7.4
file-types, v0.5.1
git-diff-details, v0.21.0
git-log, v0.4.1
gulp-snippets, v0.6.2
highlight-column, v0.5.0
highlight-line, v0.11.1
highlight-selected, v0.11.2
imdone-atom, v1.3.25
jQuery, v1.2.1
japanese-menu, v1.6.2
javascript-snippets, v1.2.1
jshint, v1.8.3
json-schema, v0.1.15
jsonlint, v1.1.2
language-babel, v2.15.3
language-django, v0.1.1
language-docker, v1.1.6
language-jade, v0.6.3
language-javascript-jsx, v0.3.7
language-markdown, v0.10.1
language-mongodb, v0.3.1
language-plantuml, v0.2.0
language-rust, v0.4.5
linter, v1.11.3
linter-bootlint, v1.0.0
linter-csslint, v1.2.0
linter-docker, v0.1.2
linter-javac, v1.5.0
linter-jscs, v3.4.5
linter-jsonlint, v1.1.5
linter-jsxhint, v0.1.2
linter-rubocop, v0.4.7
linter-ruby, v1.2.1
linter-rust, v0.3.0
markdown-format, v2.5.1
markdown-preview-plus, v2.2.2
markdown-scroll-sync, v2.1.1
markdown-table-formatter, v2.7.2
markdown-themeable-pdf, v0.10.2
minimap, v4.19.0
minimap-autohide, v0.10.1
minimap-bookmarks, v0.2.0
minimap-find-and-replace, v4.4.0
minimap-git-diff, v4.1.8
minimap-highlight-selected, v4.3.1
minimap-linter, v1.1.1
minimap-pigments, v0.1.7
minimap-selection, v4.3.1
node-debugger, v1.3.2
nodejs-snippets, v2.0.0
pdf-view, v0.39.0
pigments, v0.23.1
plantuml-viewer, v0.7.0
pretty-json, v0.4.1
preview-plus, v1.2.6
python-tools, v0.6.7
racer, v0.18.0
react, v0.14.1
react-es6-snippets, v0.3.0
react-snippets, v0.3.0
regex-railroad-diagram, v0.11.0
ruby-block, v0.3.5
rust-api-docs-helper, v0.5.1
show-ideographic-space, v1.0.1
symbols-tree-view, v0.11.0
synced-sidebar, v0.4.3
tablr, v1.0.2
tabs-to-spaces, v1.0.1
trailing-spaces, v0.3.2
tree-view-git-branch, v0.0.6
tree-view-git-status, v0.2.2
wordcount, v2.6.0

# Dev
No dev packages
```

**Comment by Bjoernsen** [comment]
[Enter steps to reproduce below:]

1. Reopen Atom
2. Switch to opened `untitled` file tab

**Atom Version**: 1.6.0-beta1
**System**: Ubuntu 14.04.2
**Thrown From**: [markdown-scroll-sync](https://github.com/mark-hahn/markdown-scroll-sync) package, v2.1.1


### Stack Trace

Uncaught TypeError: Cannot read property 'split' of undefined

```
At /home/bjoern/.atom/packages/markdown-scroll-sync/lib/main.coffee:29

TypeError: Cannot read property 'split' of undefined
    at isMarkdown (/home/bjoern/.atom/packages/markdown-scroll-sync/lib/main.coffee:29:46)
    at /home/bjoern/.atom/packages/markdown-scroll-sync/lib/main.coffee:33:10
    at Function.module.exports.Emitter.simpleDispatch (/usr/share/atom-beta/resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/usr/share/atom-beta/resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at /usr/share/atom-beta/resources/app.asar/src/pane-container.js:346:27
    at Function.module.exports.Emitter.simpleDispatch (/usr/share/atom-beta/resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/usr/share/atom-beta/resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at Pane.module.exports.Pane.setActiveItem (/usr/share/atom-beta/resources/app.asar/src/pane.js:248:22)
    at Pane.module.exports.Pane.activateItem (/usr/share/atom-beta/resources/app.asar/src/pane.js:324:21)
    at atom-tabs.TabBarView.onMouseDown (/usr/share/atom-beta/resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:596:19)
```

### Commands

```
     -5:44.3.0 core:move-right (atom-text-editor.editor.is-focused)
     -5:44.1.0 core:move-up (atom-text-editor.editor.is-focused)
     -5:43.9.0 core:select-down (atom-text-editor.editor.is-focused)
     -5:43.4.0 core:copy (atom-text-editor.editor.is-focused)
     -5:36.4.0 editor:newline (atom-text-editor.editor.is-focused)
     -5:36.2.0 core:paste (atom-text-editor.editor.is-focused)
  3x -5:11.3.0 core:copy (atom-text-editor.editor.is-focused)
  3x -2:47.3.0 core:move-right (atom-text-editor.editor.is-focused)
     -2:38.1.0 editor:move-to-end-of-word (atom-text-editor.editor.is-focused)
 10x -2:35 core:move-right (atom-text-editor.editor.is-focused)
     -2:29.7.0 core:save (atom-text-editor.editor.is-focused)
  2x -2:21.5.0 core:move-right (atom-text-editor.editor.is-focused)
  2x -2:20.8.0 core:delete (atom-text-editor.editor.is-focused)
     -2:19.1.0 core:save (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "audioBeep": false,
    "autoHideMenuBar": true,
    "themes": [
      "atom-dark-ui",
      "railscast-theme"
    ]
  }
}
```

### Installed Packages

```coffee
# User
autocomplete-emojis, v2.2.2
color-picker, v2.1.1
file-icons, v1.6.14
git-tab-status, v1.9.2
highlight-line, v0.11.1
highlight-selected, v0.11.2
language-haml, v0.23.15
linter, v1.11.3
linter-haml, v1.1.0
linter-rubocop, v0.4.7
markdown-scroll-sync, v2.1.1
minimap, v4.19.0
minimap-git-diff, v4.1.8
project-manager, v2.7.6
railscast-theme, v0.3.0

# Dev
No dev packages
```

**Comment by luckydonald** [comment]
[Enter steps to reproduce below:]

1. new blank file (<kbd>cmd</kbd>+<kbd>N</kbd>)

    >It doen't switch to it.

2. click on it

    >Error message appears, document not open.

- Also I can't close it with the little `x` button, a similar error appers.

**Atom Version**: 1.4.0
**System**: Mac OS X 10.9.5
**Thrown From**: [markdown-scroll-sync](https://github.com/mark-hahn/markdown-scroll-sync) package, v2.1.1


### Stack Trace

Uncaught TypeError: Cannot read property 'split' of undefined

```
At /Users/luckydonald/.atom/packages/markdown-scroll-sync/lib/main.coffee:29

TypeError: Cannot read property 'split' of undefined
    at isMarkdown (/Users/luckydonald/.atom/packages/markdown-scroll-sync/lib/main.coffee:29:46)
    at /Users/luckydonald/.atom/packages/markdown-scroll-sync/lib/main.coffee:33:10
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/pane-container.js:333:27
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at Pane.module.exports.Pane.setActiveItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:248:22)
    at Pane.module.exports.Pane.activateItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:316:21)
    at atom-tabs.TabBarView.onMouseDown (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:670:19)
```

### Commands

```
  3x -0:36.5.0 application:new-file (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "block-selection-mode",
      "default-encoding"
    ],
    "packagesWithKeymapsDisabled": [
      "duplicate-line-or-selection"
    ],
    "themes": [
      "one-light-ui",
      "atom-light-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
Sublime-Style-Column-Selection, v1.3.0
atom-json-editor, v0.4.1
autocomplete-json, v1.5.0
clipboard-plus, v0.5.1
color-picker, v2.1.1
column-select, v0.2.0
export-html, v1.1.0
flatten-json, v0.1.4
jshint, v1.8.3
language-arduino, v0.4.1
language-babel, v2.9.3
language-docker, v1.1.6
language-matlab-only, v0.2.0
linter-docker, v0.1.2
markdown-scroll-sync, v2.1.1
markdown-themeable-pdf, v0.10.2
minimap, v4.19.0
minimap-selection, v4.3.1
nino, v0.1.2
plist-converter, v0.1.2
pretty-json, v0.4.1
rainbow-selection, v0.3.1
react, v0.14.0
react-snippets, v0.3.0
recent-files-fuzzy-finder, v0.2.4
wakatime, v5.0.2

# Dev
No dev packages
```

**Comment by dorian-marchal** [comment]
Same issue here :+1:

**Comment by mark-hahn** [comment]
Fixed in 2.1.2.

Thanks everyone.

**Comment by dorian-marchal** [comment]
Thanks :)
